curvefs/src/client/warmup/warmup_manager.h

```diff
@@ -235,10 +273,20 @@ class WarmupManager {
      * @return true
      * @return false warmupProcess has been added
      */
-    virtual bool AddWarmupProcess(fuse_ino_t key, WarmupStorageType type) {
+    virtual bool AddWarmupProcess(fuse_ino_t key,
+                                  const std::string &path,
+                                  WarmupStorageType type) {
         WriteLockGuard lock(inode2ProgressMutex_);
-        auto ret = inode2Progress_.emplace(key, WarmupProgress(type));
-        return ret.second;
+        auto retPg = inode2Progress_.emplace(key, WarmupProgress(type));
+        auto retFp = inode2Filepath_.emplace(key, path);
+        return retPg.second && retFp.second;
+    }
+
+    virtual bool CancelWarmupProcess(fuse_ino_t key) {
+        WriteLockGuard lock(inode2ProgressMutex_);
+        auto retPg = inode2Progress_.erase(key);
+        auto retFp = inode2Filepath_.erase(key);
+        return retPg && retFp;
```

Comment:
Please sort out the warm-up process, there should be omissions in this writing

Reply:
To sum up, the data structure requires cleanup with a specific dependency sequence:
File List: `inode2Progress_` <- `warmupFilelistDeque_` <- `inode2FetchDentryPool_` -> `inode2FetchS3ObjectsPool`
File: `inode2Progress_`  <- `inode2FetchDentryPool_` <- `inode2FetchS3ObjectsPool`

My implementation is erasing the inode entry key in the sequence of `inode2Progress_`, `warmupFilelistDeque_`, `inode2FetchDentryPool_`, `inode2FetchS3ObjectsPool`

Reply:
> To sum up, the data structure requires cleanup with a specific dependency sequence:
> 
> * File List: `inode2Progress_` <- `warmupFilelistDeque_` <- `inode2FetchDentryPool_` <- `inode2FetchS3ObjectsPool`
> * File: `inode2Progress_`  <- `inode2FetchDentryPool_` <- `inode2FetchS3ObjectsPool`
> 
> [My implementation](https://github.com/opencurve/curve/pull/2652/files#diff-6d7e8dde6b6b12d5729b1c3109ea65ac32c1137f0135f3bd31aca72f74f96fd5R95) is erasing the inode entry key in the sequence of `inode2Progress_`, `warmupFilelistDeque_` (if file list), `inode2FetchDentryPool_`, `inode2FetchS3ObjectsPool`

Yes deletion should also be a top-down process

Reply:
To sum up, the data structure requires cleanup with a specific dependency sequence:

* File List: `inode2Progress_` <- `warmupFilelistDeque_` <- `inode2FetchDentryPool_` <- `warmupInodeDeque_` <- `inode2FetchS3ObjectsPool`

* File: `inode2Progress_`  <- `inode2FetchDentryPool_` <- `warmupInodeDeque_` <- `inode2FetchS3ObjectsPool`

[My implementation](https://github.com/opencurve/curve/pull/2652/files#diff-6d7e8dde6b6b12d5729b1c3109ea65ac32c1137f0135f3bd31aca72f74f96fd5R95) is erasing the inode entry key in the sequence of 
`inode2Progress_`, `warmupFilelistDeque_` (if file list), `inode2FetchDentryPool_`, `warmupInodeDeque_`, `inode2FetchS3ObjectsPool`